[2.1.0]  pointer ‘peaks0’ may be used after ‘realloc’ [-Wuse-after-free]

Hello, while preparing the Debina package of 2.1.0, I got this warning about realloc

```
src/silx/math/fit/peaks/src/peaks.c: In function ‘seek’:
src/silx/math/fit/peaks/src/peaks.c:210:36: warning: pointer ‘peaks0’ may be used after ‘realloc’ [-Wuse-after-free]
  210 |                             *peaks = peaks0;
      |                             ~~~~~~~^~~~~~~~
src/silx/math/fit/peaks/src/peaks.c:206:41: note: call to ‘realloc’ here
  206 |                         realloc_peaks = realloc(peaks0, max_npeaks * sizeof(double));
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/silx/math/fit/peaks/src/peaks.c:211:41: warning: pointer ‘relevances0’ may be used after ‘realloc’ [-Wuse-after-free]
  211 |                             *relevances = relevances0;
      |                             ~~~~~~~~~~~~^~~~~~~~~~~~~
src/silx/math/fit/peaks/src/peaks.c:207:46: note: call to ‘realloc’ here
  207 |                         realloc_relevances = realloc(relevances0, max_npeaks * sizeof(double));
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Cheers

Fred

--- a/src/silx/math/fit/peaks/src/peaks.c
+++ b/src/silx/math/fit/peaks/src/peaks.c
@@ -206,22 +206,20 @@ long seek(long begin_index,
                         realloc_peaks = realloc(peaks0, max_npeaks * sizeof(double));
                         if (realloc_peaks == NULL) {
                             printf("Error: failed to extend memory for peaks array.");
-                            *peaks = peaks0;
-                            *relevances = relevances0;
                             return(-n_peaks);
                         } else {
                             peaks0 = realloc_peaks;
+                            *peaks = peaks0;
                         }
 
                         realloc_relevances = realloc(relevances0, max_npeaks * sizeof(double));
                         if (realloc_relevances == NULL) {
                             printf("Error: failed to extend memory for peak relevances array.");
-                            *peaks = peaks0;
-                            *relevances = relevances0;
                             return(-n_peaks);
                         }
                         else {
                             relevances0 = realloc_relevances;
+                            *relevances = relevances0;
                         }
                     }
                     peaks0[n_peaks] = cch-1;
@@ -257,7 +255,5 @@ long seek(long begin_index,
         printf("index %g with y = %g\n", peaks0[i],data[(long ) peaks0[i]]);
       }
     }
-    *peaks = peaks0;
-    *relevances = relevances0;
     return (n_peaks);
 }

--- a/src/silx/math/fit/peaks/src/peaks.c
+++ b/src/silx/math/fit/peaks/src/peaks.c
@@ -1,5 +1,5 @@
 #/*##########################################################################
-# Copyright (c) 2004-2016 European Synchrotron Radiation Facility
+# Copyright (c) 2004-2024 European Synchrotron Radiation Facility
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal
@@ -204,15 +204,23 @@ long seek(long begin_index,
                     if (n_peaks == max_npeaks) {
                         max_npeaks = max_npeaks + 100;
                         realloc_peaks = realloc(peaks0, max_npeaks * sizeof(double));
-                        realloc_relevances = realloc(relevances0, max_npeaks * sizeof(double));
-                        if (realloc_peaks == NULL || realloc_relevances == NULL) {
+                        if (realloc_peaks == NULL) {
                             printf("Error: failed to extend memory for peaks array.");
+                            *peaks = peaks0;
+                            *relevances = relevances0;
+                            return(-n_peaks);
+                        } else {
+                            peaks0 = realloc_peaks;
+                        }
+
+                        realloc_relevances = realloc(relevances0, max_npeaks * sizeof(double));
+                        if (realloc_relevances == NULL) {
+                            printf("Error: failed to extend memory for peak relevances array.");
                             *peaks = peaks0;
                             *relevances = relevances0;
                             return(-n_peaks);
                         }
                         else {
-                            peaks0 = realloc_peaks;
                             relevances0 = realloc_relevances;
                         }
                     }

